legal-api/src/legal_api/services/filings/validations/incorporation_application.py

```diff
@@ -69,3 +77,67 @@ def validate_offices(incorporation_json) -> Error:
         return msg
 
     return None
+
+
+def validate_roles(incorporation_json) -> Error:
+    """Validate the required completing party of the incorporation filing."""
+    parties_array = incorporation_json['filing']['incorporationApplication']['parties']
+    msg = []
+    completing_party_count = 0
+    incorporator_count = 0
+    director_count = 0
+
+    for item in parties_array:
+        role_array = item['roles']
+
+        if role_array.count('Completing Party') == 1:
+            completing_party_count += 1
+
+        if role_array.count('Incorporator') == 1:
+            incorporator_count += 1
+
+        if role_array.count('Director') == 1:
+            director_count += 1
+
+    if completing_party_count == 0:
+        err_path = '/filing/incorporationApplication/parties/roles'
+        msg.append({'error': 'Must have a minimum of one completing party', 'path': err_path})
+
+    if completing_party_count >= 2:
+        err_path = '/filing/incorporationApplication/parties/roles'
+        msg.append({'error': 'Must have a maximum of one completing party', 'path': err_path})
+
+    if incorporator_count < 1:
+        err_path = '/filing/incorporationApplication/parties/roles'
+        msg.append({'error': 'Must have a minimum of one Incorporator', 'path': err_path})
+
+    if director_count < 1:
+        err_path = '/filing/incorporationApplication/parties/roles'
+        msg.append({'error': 'Must have a minimum of one Director', 'path': err_path})
+
+    if msg:
+        return msg
+
+    return None
+
+
+def validate_parties_mailing_address(incorporation_json) -> Error:
+    """Validate the person data of the incorporation filing."""
+    parties_array = incorporation_json['filing']['incorporationApplication']['parties']
+    msg = []
+
+    for item in parties_array:
+        for k, v in item['mailingAddress'].items():
+            if v is None:
+                err_path = '/filing/incorporationApplication/parties/mailingAddress/%s/%s/' % (
```

Comment:
If we have multiple parties with some invalid mailing address should we provide an identifier like party name or index or something to identify which fields are invalid for which party

Reply:
Great idea, will look at this!